packages/native-stack/src/types.tsx

```diff
@@ -682,6 +708,196 @@ export type NativeStackNavigationOptions = {
    */
   unstable_sheetFooter?: () => React.ReactNode;
 };
+interface SharedHeaderBarButtonItem {
+  /**
+   * Title of the button.
+   */
+  title?: string;
+  /**
+   * Style for the button title.
+   */
+  titleStyle?: {
+    fontFamily?: string;
+    fontSize?: number;
+    fontWeight?: string;
+    color?: ColorValue;
+  };
+  /**
+   * Image source for the button
+   */
+  image?: ImageRequireSource;
+  /**
+   * Any SF symbol. Explore them here: https://developer.apple.com/sf-symbols/
+   */
+  sfSymbolName?: string;
+  /**
+   * The style of the item.
+   * "Prominent" only available from iOS 26.0 and later.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uibarbuttonitem/style-swift.property
+   */
+  style?: 'plain' | 'done' | 'prominent';
+  /**
+   * The tint color to apply to the button item.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uibarbuttonitem/tintcolor
+   */
+  tintColor?: ColorValue;
+  /**
+   * A boolean that determines the visibility of the item.
+   * Only available from iOS 16.0 and later.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uibarbuttonitem/ishidden
+   */
+  hidden?: boolean;
+  /**
+   * A Boolean value that indicates whether the button is in a enabled state.
+   */
+  enabled?: boolean;
```

Comment:
This is inconsistent with conventional patterns, where you disable a button by specifying `disabled`, instead of enabling based on `enabled`.

Reply:
Same answer as https://github.com/react-navigation/react-navigation/pull/12657#discussion_r2380047036 Im using the same name as Apple (https://developer.apple.com/documentation/uikit/uibaritem/isenabled?language=objc)

Reply:
React Navigation doesn't try to match Apple-naming 1-1. This is a cross-platform library, so the API needs to make sense in a cross-platform context, not only iOS. There is no guarantee that this API won't be supported on Android or Web in the future.

The API needs to be consistent with other React Navigation APIs and the framework in general. If React Native Screens wants to keep original iOS names, then the properties should just be renamed before passing on React Navigation side.

Reply:
Got it! I will rename it asap

Reply:
https://github.com/react-navigation/react-navigation/pull/12657/commits/d3919cbcc8c769776b64c77cace86ddffcefe281